File: src/main/java/org/zenbot/timetableupdater/batch/RemoveRouteLinesExecutionListener.java
package org.zenbot.timetableupdater.batch;

import lombok.extern.slf4j.Slf4j;
import org.springframework.batch.core.JobExecution;
import org.springframework.batch.core.JobExecutionListener;
import org.springframework.stereotype.Component;
import org.zenbot.timetableupdater.dao.RouteRepository;
import org.zenbot.timetableupdater.domain.BusRoute;

import java.util.ArrayList;
import java.util.List;

@Slf4j
@Component
public class RemoveRouteLinesExecutionListener implements JobExecutionListener {

    private final RouteRepository routeRepository;

    public RemoveRouteLinesExecutionListener(RouteRepository routeRepository) {
        this.routeRepository = routeRepository;
    }

    @Override
    public void beforeJob(JobExecution jobExecution) {
        log.info("Removing all routelines from database");
        List<BusRoute> routes = routeRepository.findAll();
        routes.forEach(route -> route.setBusRouteLines(new ArrayList<>()));
        routeRepository.saveAll(routes);
    }

    @Override
    public void afterJob(JobExecution jobExecution) {
    }
}

File: src/main/kotlin/org/zenbot/timetableupdater/batch/RemoveRouteLinesExecutionListener.kt
package org.zenbot.timetableupdater.batch

import org.slf4j.LoggerFactory
import org.springframework.batch.core.JobExecution
import org.springframework.batch.core.JobExecutionListener
import org.springframework.core.env.Environment
import org.zenbot.timetableupdater.dao.RouteRepository

open class RemoveRouteLinesExecutionListener(val routeRepository: RouteRepository, val environment: Environment): JobExecutionListener {

    val log = LoggerFactory.getLogger(this::class.java)

    override fun beforeJob(jobExecution: JobExecution?) {
        val routes = routeRepository.findAll()
        val activeProfiles = environment.activeProfiles
        if (!activeProfiles.isEmpty()) {
            log.info("Removing routelines [{}]", activeProfiles.joinToString(", "))
            routes.forEach {
                busRoute -> if (activeProfiles.contains(busRoute.routename)) busRoute.busRouteLines = ArrayList()
            }
        } else {
            routes.forEach{busRoute -> busRoute.busRouteLines = ArrayList()}

        }
        routeRepository.saveAll(routes)
    }

    override fun afterJob(jobExecution: JobExecution?){}
}